index.js

```diff
@@ -169,62 +73,47 @@ function isString (argument) {
   return typeof argument === 'string'
 }
 
-function findIssues (configuration, children, dependencies, results) {
-  if (Array.isArray(children)) {
+function findIssues (configuration, children, results) {
+  if (children) {
     children.forEach(function (child) {
       if (
-        !configuration.productionOnly ||
-        appearsIn(child, dependencies)
-      ) {
-        var result = resultForPackage(configuration, child)
-        // Deduplicate.
-        var existing = results.find(function (existing) {
-          return (
-            existing.name === result.name &&
-            existing.version === result.version
-          )
-        })
-        if (existing) {
-          if (existing.duplicates) {
-            existing.duplicates.push(result)
-          } else {
-            existing.duplicates = [result]
-          }
+        configuration.productionOnly &&
+        child.dev
+      ) return
+      var result = resultForPackage(configuration, child)
+      // Deduplicate.
+      var existing = results.find(function (existing) {
+        return (
+          existing.name === result.name &&
+          existing.version === result.version
+        )
+      })
+      if (existing) {
+        if (existing.duplicates) {
+          existing.duplicates.push(result)
         } else {
-          results.push(result)
+          existing.duplicates = [result]
         }
-        findIssues(configuration, child, dependencies, results)
-      }
-      if (child.children) {
-        findIssues(configuration, child.children, dependencies, results)
+      } else {
+        results.push(result)
       }
     })
     return results
   } else return results
 }
 
-function appearsIn (installed, dependencies) {
-  var name = installed.package.name
-  var key = KEY_PREFIX + name
-  var version = installed.package.version
-  return (
-    has(dependencies, key) &&
-    dependencies[key].indexOf(version) !== -1
-  )
-}
-
 function resultForPackage (configuration, tree) {
   var packageAllowlist = configuration.packages || {}
   var result = {
-    name: tree.package.name,
+    name: tree.name,
```

Comment:
Subtle edge case here (possibly affecting the use of `tree.name` in the children selection logic above, as well): `tree.name` and `tree.package.name` are not _always_ the same.  `name` is the name by which it is referenced for `import` or `require`, the folder name in `node_modules`.  `tree.package.name` is the name as it appears in `package.json`.

For example, if you do `npm install lodash@npm:underscore@latest`, then `node.name` is `"lodash"`, but `node.package.name` is `"underscore"`.